src/reporters/udp_sender.js

```diff
@@ -93,6 +93,10 @@ export default class UDPSender {
 
     append(span: any): SenderResponse {
         let spanSize: number = this._calcSpanSize(span);
+        if (spanSize == undefined) {
+            this._logger.error(`error converting span to Thrift: ${JSON.stringify(span, null, 4)}`);
```

Comment:
definitely better than not having that error, but still doesn't explain why the conversion failed. 

The actual logic is `return this._jaegerThrift.Span.rw.byteLength(new this._jaegerThrift.Span(span)).length;` - it seems it returns a struct that has `.length` property, but if the conversion failed it probably has a more descriptive error in some other property. Maybe change _calcSpanSize to return that object and check it for error instead of `.length == undefined`